### PR TITLE
Migrate FAQ and misc content to ops section

### PR DESCRIPTION
**Pull request by john-a-joyce**
This change migrates the traffic-management FAQ
and the network related misc (previously troubleshooting)
content to the traffic-management ops guide

### CODE REVIEW

**Comment by geeknoid** [comment]
Deploy preview for *preliminary-istio* failed.

Built with commit 25a87ee7e84d4d5221b311f5aa19c15bcf41fc59

https://app.netlify.com/sites/preliminary-istio/deploys/5b5f1ec8c6aed67258a21f54

File: content/help/faq/traffic-management/ingress-with-no-route-rules.md [hunk 1]
```diff
@@ -1,57 +0,0 @@
----
-title: Can I use standard Ingress specification without any route rules?
```
Comment (geeknoid) [inline content/help/faq/traffic-management/ingress-with-no-route-rules.md]: Hmmm. The plan has been to keep this kind of content in the FAQ. The ops guide is more about day-to-day operations and troubleshooting.

File: content/help/ops/traffic-management/deploy-guidelines/index.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
+---
+title: Deployment and Configuration Guidelines
```
Comment (sdake) [inline content/help/ops/traffic-management/deploy-guidelines/index.md]: Title should only have a leading Capital -> See: https://preliminary.istio.io/about/contribute/style-guide/#only-capitalize-the-first-letter-of-headings

File: content/help/ops/traffic-management/deploy-guidelines/index.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+## 503 errors while reconfiguring service routes
+
+When setting route rules to direct traffic to specific versions (subsets) of a service, care must be taken to ensure
+that the subsets are available before they are used in the routes. Otherwise, calls to the service may return
+503 errors during a reconfiguration period.
+
+Creating both the `VirtualServices` and `DestinationRules` that define the corresponding subsets using a single `kubectl`
+call (e.g., `kubectl apply -f myVirtualServiceAndDestinationRule.yaml` is not sufficient because the
+resources propagate (from the configuration server, i.e., Kubernetes API server) to the Pilot instances in an eventually consistent manner. If the
+`VirtualService` using the subsets arrives before the `DestinationRule` where the subsets are defined, the Envoy configuration generated by Pilot would refer to non-existent upstream pools, resulting in HTTP 503 errors until all configuration objects are available to Pilot.
```
Comment (sdake) [inline content/help/ops/traffic-management/deploy-guidelines/index.md]: RO.  terminate pools, with pools.  Resulting in -> This scenario results in

File: content/help/ops/traffic-management/deploy-guidelines/index.md [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+To make sure services will have zero down-time when configuring routes with subsets, follow a "make-before-break" process as described below:
+
+* When adding new subsets:
+
+    1. Update `DestinationRules` to add a new subset first, before updating any `VirtualServices` that use it. Apply the rule using `kubectl` or any platform-specific tooling.
+
+    1. Wait a few seconds for the `DestinationRule` configuration to propagate to the Envoys
```
Comment (sdake) [inline content/help/ops/traffic-management/deploy-guidelines/index.md]: Envoys -> Envoy side cars.

File: content/help/ops/traffic-management/deploy-guidelines/index.md [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+    1. Update the `VirtualService` to refer to the newly added subsets.
+
+* When removing subsets:
+
+    1. Update `VirtualServices` to remove any references to a subset, before removing the subset from a `DestinationRule`.
+
+    1. Wait a few seconds for the `VirtualService` configuration to propagate to the Envoys
```
Comment (sdake) [inline content/help/ops/traffic-management/deploy-guidelines/index.md]: Envoys -> Envoy side cars.

Comment (sdake) [reply content/help/ops/traffic-management/deploy-guidelines/index.md]: Another option is to use language such as "propagate within the Envoy data plane."

File: content/help/ops/traffic-management/deploy-guidelines/index.md [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+
+    1. Update the `DestinationRule` to remove the unused subsets
```
Comment (sdake) [inline content/help/ops/traffic-management/deploy-guidelines/index.md]: need a terminating period.

File: content/help/ops/traffic-management/deploy-guidelines/index.md [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+
+## Route rules have no effect on ingress gateway requests
+
+Let's assume you are using an ingress `Gateway` and corresponding `VirtualSerive` to access an internal service.
```
Comment (sdake) [inline content/help/ops/traffic-management/deploy-guidelines/index.md]: Let's assume -> If

File: content/help/ops/traffic-management/deploy-guidelines/index.md [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+For example, your `VirtualService` looks something like this:
```
Comment (sdake) [inline content/help/ops/traffic-management/deploy-guidelines/index.md]: remove period on 40 -> In this example,

File: content/help/ops/traffic-management/deploy-guidelines/index.md [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+  - myapp-gateway
+  http:
+  - match:
+    - uri:
+        prefix: /hello
+    route:
+    - destination:
+        host: helloworld.default.svc.cluster.local
+  - match:
+    ...
```
Comment (sdake) [inline content/help/ops/traffic-management/deploy-guidelines/index.md]: if someone C&Ps the ... - may result in confusion

File: content/help/ops/traffic-management/deploy-guidelines/index.md [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+  - helloworld.default.svc.cluster.local
+  http:
+  - route:
+    - destination:
+        host: helloworld.default.svc.cluster.local
+        subset: v1
+{{< /text >}}
+
+In this situation you will notice that requests to the helloworld service via the ingress gateway will
+not be directed to subset v1 but instead will continue to use default round-robin routing.
```
Comment (sdake) [inline content/help/ops/traffic-management/deploy-guidelines/index.md]: v1 may need `

File: content/help/ops/traffic-management/deploy-guidelines/index.md [hunk 11]
```diff
@@ -0,0 +1,205 @@
...
+
+The ingress requests are using the gateway host (e.g., `myapp.com`)
+which will activate the rules in the myapp `VirtualService` that routes to any endpoint in the helloworld service.
+On the other hand, internal requests with host `helloworld.default.svc.cluster.local` will use the
```
Comment (sdake) [inline content/help/ops/traffic-management/deploy-guidelines/index.md]: Avoid idioms such as "On the other hand"

File: content/help/ops/traffic-management/deploy-guidelines/index.md [hunk 12]
```diff
@@ -0,0 +1,205 @@
...
+
+## Route rules have no effect on my application
+
+If route rules are working perfectly for the [Bookinfo](/docs/examples/bookinfo/) sample,
+but similar version routing rules have no effect on your own application, it may be that
+your Kubernetes services need to be changed slightly.
+
+Kubernetes services must adhere to certain restrictions in order to take advantage of
+Istio's L7 routing features.
+Refer to the [Requirements for Pods and Services](/docs/setup/kubernetes/spec-requirements)
```
Comment (sdake) [inline content/help/ops/traffic-management/deploy-guidelines/index.md]: Pods and Services should be lowercase.  If spec-requirements has that as uppercase, we merged an error.  Can you address?

File: content/help/ops/traffic-management/deploy-guidelines/index.md [hunk 13]
```diff
@@ -0,0 +1,205 @@
...
+for details.
+
+## Envoy won't connect to my HTTP/1.0 service
+
+Envoy requires HTTP/1.1 or HTTP/2 traffic for upstream services. For example, when using [NGINX](https://www.nginx.com/) for serving traffic behind Envoy, you
```
Comment (sdake) [inline content/help/ops/traffic-management/deploy-guidelines/index.md]: HTTP/1.1 and HTTP/2 will need backticks

File: content/help/ops/traffic-management/deploy-guidelines/index.md [hunk 14]
```diff
@@ -0,0 +1,205 @@
...
+will need to set the [proxy_http_version](https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_http_version) directive in your NGINX configuration to be "1.1", since the NGINX default is 1.0
```
Comment (sdake) [inline content/help/ops/traffic-management/deploy-guidelines/index.md]: need terminating period.

File: content/help/ops/traffic-management/deploy-guidelines/index.md [hunk 15]
```diff
@@ -0,0 +1,205 @@
...
+    location /http/ {
+        proxy_pass http://http_backend;
+        proxy_http_version 1.1;
+        proxy_set_header Connection "";
+        ...
+    }
+}
+{{< /text >}}
+
+## Headless TCP services losing connection from Istiofied containers
```
Comment (sdake) [inline content/help/ops/traffic-management/deploy-guidelines/index.md]: Istiofied->Istio managed

File: content/help/ops/traffic-management/deploy-guidelines/index.md [hunk 16]
```diff
@@ -0,0 +1,205 @@
...
+
+If `istio-citadel` is deployed, Envoy is restarted every 15 minutes to refresh certificates.
+This causes the disconnection of TCP streams or long-running connections between services.
+
+You should build resilience into your application for this type of
+disconnect, but if you still want to prevent the disconnects from
+happening, you will need to disable mutual TLS and the `istio-citadel` deployment.
```
Comment (sdake) [inline content/help/ops/traffic-management/deploy-guidelines/index.md]: I think you still need istio-citadel even with Mutual TLS is disabled...  @myidpt comments?

File: content/docs/setup/kubernetes/spec-requirements/index.md [hunk 17]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Requirements for Pods and Services
+title: Requirements for pods and services
```
Comment (geeknoid) [inline content/docs/setup/kubernetes/spec-requirements/index.md]: Actually, contrary to what Steve said, titles should have full capitalization.

Section headers (the ## stuff) is what uses initial-caps only.

I'll clarify that in the guidelines.

Comment (sdake) [reply content/docs/setup/kubernetes/spec-requirements/index.md]: @geeknoid my bad :)

File: content/help/ops/traffic-management/deploy-guidelines/index.md (hunk 11 above)
Comment (sdake) [reply content/help/ops/traffic-management/deploy-guidelines/index.md]: Replace "On the other hand, internal requests with host" with "With internal requests containing the host"

File: content/help/ops/traffic-management/observing/index.md [hunk 18]
```diff
@@ -1,11 +1,32 @@
...
+[2017-05-17 03:00:52.735][14236][critical][assert] assert failure: fd_ != -1: external/envoy/source/common/network/connection_impl.cc:58
+{{< /text >}}
+
+Make sure to raise your ulimit. Example: `ulimit -n 16384`
+
+## Why is creating a weighted route rule to split traffic between two versions of a service not working as expected?
+
+For the current Envoy sidecar implementation, up to 100 requests may be required for the desired distribution to be observed.
+
+## How come some of my route rules don't take affect immediately?
```
Comment (sdake) [inline content/help/ops/traffic-management/observing/index.md]: affect vs effect - blows peoples minds and sticks out - can you use impact instead?

Cheers
-steve

File: content/help/ops/traffic-management/deploy-guidelines/index.md (hunk 2 above)
Comment (sdake) [reply content/help/ops/traffic-management/deploy-guidelines/index.md]: ignore

File: content/help/ops/traffic-management/deploy-guidelines/index.md (hunk 10 above)
Comment (sdake) [reply content/help/ops/traffic-management/deploy-guidelines/index.md]: ignore

**Pull request merged**